Use BDurationFormat to format time intervals and display alert for similar task names.

AkshayAgarwal007 opened the pull request:
* Use BDurationFormat to format time intervals.
* Display an alert in case the user enters identical task name.
* Minor cleanups (Use proper variable name)
* Add gitignore.

Closes #12.

waddlesplash commented on src/TimeTrackerWindow.cpp:
```diff
@@ -106,14 +106,33 @@ TimeTrackerWindow::MessageReceived(BMessage* message)
-		break;
+		TaskName.SetTo(message->FindString("Title"));
+
+		int32 button_inndex;
```
`index` not `inndex`.

waddlesplash commented on src/TimeTrackerWindow.cpp:
```diff
@@ -106,14 +106,33 @@ TimeTrackerWindow::MessageReceived(BMessage* message)
+			}
+		}
+
+		if (button_inndex==0) {
```
Spaces around `==`.

waddlesplash commented:
Looks OK other than comments.

AkshayAgarwal007 committed:
Display alert for similar task names and remove 'Days-0' in spent time.

Display an alert in case the user enters indentical task name, removed '0-days' passed in spent time, show days passed after 1 day had passed. Closes #12.

AkshayAgarwal007 commented:
done!

pulkomandy commented on src/TaskListItem.cpp:
```diff
@@ -135,7 +135,12 @@ TaskListItem::GetStringTime()
-	sprintf(buffer, "%d days, %d:%d:%d", Day, Hours, Minutes, Seconds);
+	if (Day==0)
+		sprintf(buffer, "%d:%d:%d", Hours, Minutes, Seconds);
+	else if (Day==1)
```
Such checks are not appropriate for any language other than english. For example, in Russian: https://github.com/HaikuArchives/TimeTracker/pull/17 there are different rules for 2 to 4, than for 5 and up.

I think the best thing to do here is to use a BDurationFormat, which will do the right thing for you. For an example use, you can check AboutSystem in Haiku, which uses this to show the system uptime.

pulkomandy commented on src/TimeTrackerWindow.cpp:
```diff
@@ -106,14 +106,33 @@ TimeTrackerWindow::MessageReceived(BMessage* message)
-		break;
+		TaskName.SetTo(message->FindString("Title"));
+
+		int32 button_index;
```
You need to initialize this to 0, otherwise, in the case where there are no other tasks with the same name, it is never set and the if to decide wether to add the new task will have undefined result.

AkshayAgarwal007 replied:
Worked while checking it multiple times probably because the garbage value was 0.

pulkomandy reviewed (changes requested):
Looks good overall, but please fix these two small issues.

AkshayAgarwal007 committed:
Added gitignore and minor cleanups

AkshayAgarwal007 committed:
Use BDurationFormat class to format time intervals and bug fixes

AkshayAgarwal007 commented:
Screenshot with abbreviated style formatting.

![timetracker-new](https://cloud.githubusercontent.com/assets/8170214/26175484/6b2a2466-3b70-11e7-9861-a0ce883ec76f.png)

AkshayAgarwal007 committed:
Move style flag to constructor call as per API changes.

scottmc commented:
Can someone fix up the conflicts and merge this?  Looks like it was ready but we forgot to merge it?

waddlesplash committed:
Merge branch 'master' into cleanups
